### DISCUSSION TITLE
"@export_category"-like way to add catergories to shader parameters on a material

### DISCUSSION THREAD

**Discussion by FireCatMagic** [godotengine/godot-proposals#11146] [Engine Core]
![image](https://github.com/user-attachments/assets/68e60b1c-62b6-4788-bc3e-625d303a20dc)
Basically, would accept a string and allow a expandable catergory like how you can in GDScript

I'm not sure of the implementation of this because GDShader isn't GDScript, it does not have annotations

**Comment by Jesusemora** [comment]
you can already do this.
The `group_uniforms` keyword.

```
group_uniforms misc;
uniform vec2 uv_scale = vec2(1.0, 1.0);
```

**Comment by FireCatMagic** [comment]
oh 😭 sorry
